cyberdrop_dl/managers/download_manager.py

```diff
@@ -69,9 +70,12 @@ async def basic_auth(self, username, password) -> str:
         token = b64encode(f"{username}:{password}".encode('utf-8')).decode("ascii")
         return f'Basic {token}'
 
-    async def check_free_space(self) -> bool:
+    async def check_free_space(self, folder: Path = None) -> bool:
         """Checks if there is enough free space on the drive to continue operating"""
-        free_space = shutil.disk_usage(self.manager.path_manager.download_dir.parent).free
+        if folder is None:
+            free_space = disk_usage(self.manager.path_manager.download_dir).free
+        else:
+            free_space = disk_usage(folder).free
```

Comment:
Tested but `check_free_space` errors out if `folder` does not exists.

**Logs**

[downloader.log](https://github.com/user-attachments/files/17091588/downloader.log)

Reply:
Reviewing the code downstream seems that the reason Jules originally used `self.manager.path_manager.download_dir.parent` instead of just `self.manager.path_manager.download_dir` was because at this point in the code `download_dir` has not been created yet.

Reply:
Would switching it to `self.manager.path_manager.download_dir.drive` work? I'm not sure if it would work on a ramdisk.